models/repo_tree.go

```diff
@@ -0,0 +1,79 @@
+// TODO:
+// With model objects here, we can write tests for them and for TreeListing? Or maybe
+// they should go elsewhere?
+
+// Then we can write fuller integration tests, with model objects that we're expecting and asserting
+// against
+import (
+	"path/filepath"
+
+	"code.gitea.io/git"
+)
+
+// RepoFile represents a file blob contained in the repository
+type RepoFile struct {
+	Path     string            `json:"path"`
+	// Mode     git.EntryMode     `json:"mode"`  // TODO: Do we include this? It'll require exporting the mode as public in the `git` module...
```

Comment:
How does GitHub API represent them? usually we create a `APIFormat` function for things like this